screens/SignUpScreen.js

```diff
@@ -1,40 +1,65 @@
 import { useNavigation } from '@react-navigation/native';
-import { ImageBackground } from 'react-native';
+import { Alert, ImageBackground } from 'react-native';
 import { StyleSheet, Text, View } from 'react-native';
 import Input from '../components/Auth/Input';
 import SecondaryButton from '../components/ui/SecondaryButton';
 import { useSelector, useDispatch } from 'react-redux';
 import {
+  clearUserInputFields,
   onBodyChangeEmail,
   onBodyChangeFirstName,
   onBodyChangeLastName,
   onBodyChangePassword,
 } from '../redux/userSlice';
-import { signUpUser } from '../components/Auth/Services/client';
+import {
+  getUser,
+  signInUser,
+  signUpUser,
+} from '../components/Auth/Services/client';
+import {
+  authenticateUser,
+  setAuthenticatedUser,
+  setUserAuthenticationToken,
+} from '../redux/authSlice';
 
 export default function SignUpScreen() {
+  const navigation = useNavigation();
   const { email, firstName, lastName, password } = useSelector(
     (state) => state.user
   );
   const dispatch = useDispatch();
 
   async function submitHandler() {
-    const newUser = {
-      email: email,
-      firstName: firstName,
-      lastName: lastName,
-      password: password,
-    };
-    await signUpUser(newUser);
-    console.log(newUser);
-    dispatch(onBodyChangeEmail(''));
-    dispatch(onBodyChangeFirstName(''));
-    dispatch(onBodyChangeLastName(''));
-    dispatch(onBodyChangePassword(''));
-    return;
+    const validEmail = email.includes('@');
```

Comment:
Would `@@@@` be a valid email then? You might want to do more robust email validation here. There are npm packages and pre-made regex to handle this.

Reply:
noted. For now in development, I'm thinking of keeping it simple for the sake of creating users/ logging in and out. I'll add in more validation towards the end. Thanks for pointing that out, I completely overlooked those potential options